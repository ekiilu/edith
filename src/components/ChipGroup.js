import React from "react"

import { makeStyles } from "@material-ui/core/styles"
import Chip from "@material-ui/core/Chip"

const useStyles = makeStyles(theme => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    "& > *": {
      margin: theme.spacing(0.2),
    },
  },
}))

/**
 * ChipGroup displays a horizontal group of Chips.
 */
export default props => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      {props.labels?.map((label, index) => (
        <Chip key={index} size="small" label={label} />
      ))}
    </div>
  )
}
