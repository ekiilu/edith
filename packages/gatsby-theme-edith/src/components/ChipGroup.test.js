import React from "react"
import { render } from "@testing-library/react"

import ChipGroup from "./ChipGroup"

describe("ChipGroup", () => {
  it("renders correctly with no labels", () => {
    const { asFragment } = render(<ChipGroup />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with one label", () => {
    const labels = ["Foo"]
    const { asFragment } = render(<ChipGroup labels={labels} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with multiple labels", () => {
    const labels = ["Foo", "Bar", "Baz"]
    const { asFragment } = render(<ChipGroup labels={labels} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
