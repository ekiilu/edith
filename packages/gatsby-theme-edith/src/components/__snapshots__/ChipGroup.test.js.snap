// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipGroup renders correctly with multiple labels 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-33"
  >
    <div
      class="MuiChip-root"
    >
      <span
        class="MuiChip-label"
      >
        Foo
      </span>
    </div>
    <div
      class="MuiChip-root"
    >
      <span
        class="MuiChip-label"
      >
        Bar
      </span>
    </div>
    <div
      class="MuiChip-root"
    >
      <span
        class="MuiChip-label"
      >
        Baz
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChipGroup renders correctly with no labels 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  />
</DocumentFragment>
`;

exports[`ChipGroup renders correctly with one label 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-2"
  >
    <div
      class="MuiChip-root"
    >
      <span
        class="MuiChip-label"
      >
        Foo
      </span>
    </div>
  </div>
</DocumentFragment>
`;
